app/src/main/java/com/naxtlevelofandroiddevelopment/marvelgallery/network/MarvelRepository.kt

```diff
@@ -0,0 +1,28 @@
+package com.naxtlevelofandroiddevelopment.marvelgallery.network
+
+import com.naxtlevelofandroiddevelopment.marvelgallery.model.MarvelCharacter
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.providers.Provider
+import com.naxtlevelofandroiddevelopment.marvelgallery.network.providers.RetrofitApi
+import io.reactivex.Single
+
+interface MarvelRepository {
+
+    fun getCharacters(searchQuery: String?): Single<List<MarvelCharacter>>
+
+    class Impl(val api: MarvelApi) : MarvelRepository {
```

Comment:
This way we can easily override it for unit tests:
```
MarvelRepository.override = object: MarvelRepository {
    fun getCharacters(searchQuery: String?): Single<List<MarvelCharacter>> = //...
}
```

We have simple dependency injection:
```
val repository by MarvelRepository.lazyGet()
```

And if we would like to make multi-platform implementation (+ iOS thanks to Kotlin\Native) then we can easily move implementation code to Android module, write in iOS module network repository implementation for iOS and leave this in common module without any changes in its usage.